### PR TITLE
SO-4642: Change module of inactivated content

### PR DESCRIPTION
**Pull request by molnarlaura**

### CODE REVIEW

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/change/ComponentInactivationChangeProcessor.java [hunk 1]
```diff
@@ -93,13 +91,21 @@ public void process(StagingArea staging, RevisionSearcher searcher) throws IOExc
...
+		
 		if (!inactivatedConceptIds.isEmpty()) {
 			
+			
 			Multimap<String, RevisionDiff> changedMembersByReferencedComponentId = HashMultimap.create();
 			staging.getChangedRevisions(SnomedRefSetMemberIndexEntry.class).forEach(diff -> {
 				changedMembersByReferencedComponentId.put(((SnomedRefSetMemberIndexEntry) diff.newRevision).getReferencedComponentId(), diff);
 			});
 			
+			SnomedRefSetMemberIndexEntry memberForModule = staging.getChangedRevisions(SnomedRefSetMemberIndexEntry.class).findFirst().map(diff -> (SnomedRefSetMemberIndexEntry) diff.newRevision).get();
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/change/ComponentInactivationChangeProcessor.java]: It seems like this member can come from anywhere. What is its relation to the description inactivation member we are going to use `moduleId` with?

**Comment by apeteri** [comment]
Unfortunately `snomed-e2e-tests` are failing on this branch (both here on GitHub and after merging in the latest changes from 7.x, locally). The primary issue is that some transaction contexts in the SNOMED CT store do not provide an instance of `ModuleIdProvider`, like the RF2 importer and code system creation commits, including the creation of the primary code system in the terminology repository initializer. 

Tests can not start without this resolved.

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/components/SnomedConceptApiTest.java [hunk 2]
```diff
@@ -945,4 +927,51 @@ public void testPendingMoveUpdateOnConceptWithSetDescriptions() {
 		});
 	}
 	
+	@Test
+	public void testConceptInactivationModuleChanges() {
+		final String conceptId = createNewConcept(branchPath);
+		String sourceRelationshipId = createNewRelationship(branchPath, conceptId, Concepts.HAS_DOSE_FORM, Concepts.MODULE_SCT_MODEL_COMPONENT);
+		String destinationRelationshipId = createNewRelationship(branchPath, Concepts.MODULE_SCT_MODEL_COMPONENT, Concepts.HAS_DOSE_FORM, conceptId);
+		CodeSystemURI codeSystemURI = CodeSystemURI.branch(SnomedTerminologyComponentConstants.SNOMED_SHORT_NAME, branchPath.getPath());
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/components/SnomedConceptApiTest.java]: `branchPath` is absolute, so this test looks for a branch path named `MAIN/MAIN/...`, and so can not find what it was looking for.

Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/components/SnomedConceptApiTest.java]: Done

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/change/ComponentInactivationChangeProcessor.java (hunk 1 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/change/ComponentInactivationChangeProcessor.java]: As discussed, I leave the code as it is.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/change/ComponentInactivationChangeProcessor.java [hunk 3]
```diff
@@ -93,23 +91,28 @@ public void process(StagingArea staging, RevisionSearcher searcher) throws IOExc
...
 			Multimap<String, RevisionDiff> changedMembersByReferencedComponentId = HashMultimap.create();
 			staging.getChangedRevisions(SnomedRefSetMemberIndexEntry.class).forEach(diff -> {
 				changedMembersByReferencedComponentId.put(((SnomedRefSetMemberIndexEntry) diff.newRevision).getReferencedComponentId(), diff);
 			});
 			
-			for (Hits<String[]> hits : searcher.scroll(Query.select(String[].class)
+			for (Hits<SnomedDescriptionIndexEntry> hits : searcher.scroll(Query.select(SnomedDescriptionIndexEntry.class)
 					.from(SnomedDescriptionIndexEntry.class)
-					.fields(SnomedDescriptionIndexEntry.Fields.ID, SnomedDescriptionIndexEntry.Fields.MODULE_ID)
+					.fields(SnomedDescriptionIndexEntry.Fields.ID)
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/change/ComponentInactivationChangeProcessor.java]: `MODULE_ID` is still needed so the provider can work :)

**Review by cmark — APPROVED** [review]
LGTM!

**Pull request merged**